statsmodels/sandbox/regression/gmm.py

```diff
@@ -1486,6 +1486,60 @@ def predict(self, params, exog=None):
 
         return np.dot(exog, params)
 
+    def compare_j(self, weights=None, var=None):
+        # The function computes the C statistic which compare the two nested linear GMM
+        # models. We use the same weight matrix so that the difference is always
+        # positive.
+        # var takes a list giving the position of the variable to be
+        # removed in the second regression. Example: var=[2,5,8]
```

Comment:
I'm not sure, I'm thinking more about the general case than LinearIV

AFAIR, we use j-test to
- test overidentifying restrictions
- test a constrained model

i.e. we can impose constraints like zero-constraints on parameters or we can drop moment conditions while keeping the number of estimated parameters unchanged.

(I got confused about this once before, but didn't figure out which way to go with something similar. 
https://github.com/josef-pkt/misc/blob/gmm_glm/notebooks/ex_gmm_glm.ipynb
IIRC I tried to get a LM or conditional moment test where the model is evaluated under the null constraints and we want to test whether an additional moment condition holds.

This was 4 years ago, since then I figured out more of score tests and conditional moment tests.)
)